canvas.py

```diff
@@ -1,94 +1,114 @@
+        self.bottom_button = Button(text="Bottom", pos_hint={'x': .45, 'top': .2}, size_hint=(.1, .1), id='bottom')
+        self.label = Label(text="Center", pos_hint={'x': .45, 'top': .3}, size_hint=(.1, .1))
+        self.right_button = Button(text="Right", pos_hint={'x': .55, 'top': .3}, size_hint=(.1, .1), id='right')
+        self.clear_button = Button(text="Clear", pos_hint={'x': .9, 'top': .3}, size_hint=(.05, .05), id='clear')
 
         self.add_widget(parent)
         self.add_widget(self.clear_button)
         self.add_widget(self.left_button)
         self.add_widget(self.top_button)
         self.add_widget(self.bottom_button)
-        self.add_widget(self.center_button)
+        self.add_widget(self.label)
         self.add_widget(self.right_button)
 
         self.clear_button.bind(on_release=self.clear_canvas)
-        self.left_button.bind(on_press=self.button_click)
-        self.top_button.bind(on_press=self.button_click)
-        self.bottom_button.bind(on_press=self.button_click)
-        self.right_button.bind(on_press=self.button_click)
-
-        #Uses partial to create an event for the button bind and to pass token value to the Canvas class depending on button press
-        callback0 = partial(self.painter.paint, 0)
-        callback1 = partial(self.painter.paint, 1)
-        callback2 = partial(self.painter.paint, 2)
-        callback3 = partial(self.painter.paint, 3)
-
-        self.left_button.bind(on_press=callback0)
-        self.right_button.bind(on_press=callback1)
-        self.top_button.bind(on_press=callback2)
-        self.bottom_button.bind(on_press=callback3)
-
-    
+        self.left_button.bind(on_press=self._on_press, on_release=self._on_release)
+        self.top_button.bind(on_press=self._on_press, on_release=self._on_release)
+        self.bottom_button.bind(on_press=self._on_press, on_release=self._on_release)
+        self.right_button.bind(on_press=self._on_press, on_release=self._on_release)
+
     def clear_canvas(self, obj):
         self.painter.canvas.clear()
-        global X,Y 
+        global X, Y
         X = 400
         Y = 400
-    
-    def button_click(self, event):
-        self.center_button.text = event.text
 
+    def _on_press(self, event):
+        """
+        On button press a job is scheduled to paint the canvas, then the center button text is changed
+        :param event: button event
+        """
+        self.button_event = Clock.schedule_interval(partial(self.paint_canvas, event), 0.05)
+        self.label.text = event.text
+
+    def _on_release(self, event):
+        """
+        On button release scheduled paint job is cancelled
+        :param event: button event
+        """
+        Clock.unschedule(self.button_event)
+
+    def on_touch_up(self, touch):
+        """
+        Used in conjunction with Button on_release functionality
+        :param touch: event
+        """
+        Clock.unschedule(self.button_event)
+
+    def paint_canvas(self, event, dt):
```

Comment:
This def method isn't really necessary, it just adds comments to the def on the class above. Instead, maybe comment the above def ?